pkg/manager/member/tidb_member_manager.go

```diff
@@ -389,6 +389,7 @@ func (tmm *tidbMemberManager) syncTidbClusterStatus(tc *v1alpha1.TidbCluster, se
 		oldTidbMember, exist := tc.Status.TiDB.Members[name]
 		if exist {
 			newTidbMember.LastTransitionTime = oldTidbMember.LastTransitionTime
+			newTidbMember.NodeName = oldTidbMember.NodeName
```

Comment:
copy the old value because we should not clear it in any case (e.g. new pod is in pending status)